litellm/llms/ollama_chat.py

```diff
@@ -121,6 +123,63 @@ def get_config(cls):
         }
 
 
+class MessageJSON(TypedDict):
+    role: str
+    content: str
+
+
+class ChatResponseJSON(TypedDict, total=False):
+    model: str
+    created_at: str
+    message: MessageJSON
+    done: bool
+    total_duration: int
+    load_duration: int
+    prompt_eval_count: int
+    prompt_eval_duration: int
+    eval_count: int
+    eval_duration: int
+
+
+# Usage metrics are only populated when the ollama response indicates `"done": true`
+# https://github.com/jmorganca/ollama/blob/main/docs/api.md#generate-a-chat-completion
+class OllamaChatUsage:
+    def __init__(self, messages: List[MessageJSON], response_json: ChatResponseJSON):
+        self.messages = messages
+        self.response_json = response_json
+
+    def get_usage(self):
+        if self._done() == False:
+            return litellm.Usage(
+                prompt_tokens=None,
+                completion_tokens=None,
+                total_tokens=None,
+            )
+
+        # metrics are sometimes missing from the response, so also use a fallback mechanism
+        self.prompt_tokens = self.response_json.get("prompt_eval_count", self._prompt_tokens_fallback())
+        self.completion_tokens = self.response_json.get("eval_count", self._completion_tokens_fallback())
+
+        return litellm.Usage(
+            prompt_tokens=self.prompt_tokens,
+            completion_tokens=self.completion_tokens,
+            total_tokens=self.prompt_tokens + self.completion_tokens,
+        )
+
+
+    def _done(self):
+       return self.response_json["done"] == True
+
+    def _prompt_tokens_fallback(self):
+        print_verbose(f"Warning: `prompt_eval_count` missing from response, estimating by using OpenAI token counting with cl100k_base encoding.")
+        return openai_token_counter(messages=self.messages)
```

Comment:
Token counting is a lot trickier for the `/chat/completions` endpoint. I figured that we should just reuse the `openai_token_counter` for the fallback and provide a warning message?